### PR TITLE
pts undefined test unit

### PR DESCRIPTION
**Pull request by vdeshpande**
first commit - apologies if things are messy

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ module.exports = function(grunt) {
               'src/playlist-loader.js',
               'node_modules/pkcs7/dist/pkcs7.unpad.js',
               'src/decrypter.js'
+
```
Comment (dmlap) [inline Gruntfile.js]: Accidental whitespace?

File: example.html [hunk 2]
```diff
@@ -54,12 +54,18 @@
       padding: 0 5px;
       margin: 20px 0;
     }
+	.video-js
```
Comment (dmlap) [inline example.html]: Looks like you may be using tab-based indentation. This project uses 2-space indent

File: example.html [hunk 3]
```diff
@@ -54,12 +54,18 @@
...
+		height: 360px;    
+		}
+		
   </style>
 
 </head>
 <body>
   <div class="info">
-    <p>The video below is an <a href="https://developer.apple.com/library/ios/documentation/networkinginternet/conceptual/streamingmediaguide/Introduction/Introduction.html#//apple_ref/doc/uid/TP40008332-CH1-SW1">HTTP Live Stream</a>. On desktop browsers other than Safari, the HLS plugin will polyfill support for the format on top of the video.js Flash tech.</p>
+    <p>The video below is an <a href="https://developer.apple.com/library/ios/documentation/networkinginternet/conceptual/streamingmediaguide/Introduction/Introduction.html#//apple_ref/doc/uid/TP50008332-CH1-SW1">HTTP Live Stream</a>. On desktop browsers other than Safari, the HLS plugin will polyfill support for the format on top of the video.js Flash tech.</p>
```
Comment (dmlap) [inline example.html]: Why the change in this URL?

File: src/videojs-hls.js [hunk 4]
```diff
@@ -3,688 +3,518 @@
  * The main file for the HLS project.
  * License: https://github.com/videojs/videojs-contrib-hls/blob/master/LICENSE
  */
+ /*jshint -W092 */
+ /*jshint -W062 */
```
Comment (dmlap) [inline src/videojs-hls.js]: Why add these two jshint overrides? What are they?

File: src/videojs-hls.js [hunk 5]
```diff
@@ -3,688 +3,518 @@
...
 (function(window, videojs, document, undefined) {
-'use strict';
 
-var
```
Comment (dmlap) [inline src/videojs-hls.js]: I think these are all whitespace changes. If so, they should be removed from this PR.

File: test/videojs-hls_test.js [hunk 6]
```diff
@@ -2651,4 +2651,19 @@ test('does not download segments if preload option set to none', function() {
   strictEqual(loadedSegments, 0, 'did not download any segments');
 });
 
+test('pts value undefined test', function(){
```
Comment (dmlap) [inline test/videojs-hls_test.js]: Can you be a bit more descriptive?

File: test/videojs-hls_test.js [hunk 7]
```diff
@@ -2651,4 +2651,19 @@ test('does not download segments if preload option set to none', function() {
...
+	//Making a fictious array
+    var tags = [{ pts: 1000, bytes: new Uint8Array(1)}];
+	//Check the length of tags
+	ok(tags.length,"We have the length of tags");
+	//Check to see if value of pts exists
+	ok(tags[length].pts,"We have got a pts value at length of tag");
```
Comment (dmlap) [inline test/videojs-hls_test.js]: Where is `length` defined?

File: test/videojs-hls_test.js [hunk 8]
```diff
@@ -2651,4 +2651,19 @@ test('does not download segments if preload option set to none', function() {
...
+	//Check to see if the last element is not undefined
+	notEqual(tags[length].pts,undefined,"the tags array is doing fine until the last pts value");
+	//Check to see if the out of bounds array is undefined
+	equal(tags[length+1],undefined,"the tags array will going out of bounds here");
```
Comment (dmlap) [inline test/videojs-hls_test.js]: This test doesn't exercise any of contrib-hls. You want to show that when seeking, we can iterate off the end of the parsed tags if the target PTS value is greater than the PTS value of the latest tag. You'll probably need to involve the [mockSegmentParser](https://github.com/vdeshpande/videojs-contrib-hls/blob/pull-request-vineet/test/videojs-hls_test.js#L1479) and an instance of the HLS tech itself.

File: Gruntfile.js (hunk 1 above)
Comment (vdeshpande) [inline Gruntfile.js]: yes, will fix that. thanks!

File: example.html (hunk 3 above)
Comment (vdeshpande) [inline example.html]: This change is accidental

File: src/videojs-hls.js (hunk 4 above)
Comment (vdeshpande) [inline src/videojs-hls.js]: I was adding these overrides because of the jshint error I was getting while grunting the repo. Mixed spaces in the repo.
Do these overrides cause any issues ? Should I remove ?

File: src/videojs-hls.js (hunk 5 above)
Comment (vdeshpande) [inline src/videojs-hls.js]: Will do that.

File: test/videojs-hls_test.js (hunk 6 above)
Comment (vdeshpande) [inline test/videojs-hls_test.js]: I'll rename the function. Should I be very descriptive or just an in short description will do ?

File: test/videojs-hls_test.js (hunk 7 above)
Comment (vdeshpande) [inline test/videojs-hls_test.js]: I'm just getting the length of the fictitous array

File: src/videojs-hls.js (hunk 4 above)
Comment (dmlap) [inline src/videojs-hls.js]: Yes, please remove the overrides. The hint was telling you not to use tabs for indentation :)

File: test/videojs-hls_test.js [hunk 9]
```diff
@@ -2651,4 +2650,26 @@ test('does not download segments if preload option set to none', function() {
...
+    player.src({
+    src:'manifest/playlist.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+  openMediaSource(player);
+  player.trigger('seeked');
+	//Check the length of tags
+	ok(tags.length,'We have the length of tags');
+	//Check to see if value of pts exists
+	ok(tags[length].pts,'We have got a pts value at length of tag');
```
Comment (gkatsev) [inline test/videojs-hls_test.js]: where is `length` defined?

**Comment by gkatsev** [comment]
Style issues that would need to be addressed as well :)

* [ ] extra whitespace
* [ ] removed vertical whitespace
* [ ] multi-line variables squashed into one line
* [ ] comments should be on the preceding line and not on the same line
* [ ] url change
* [ ] consistent indentation (2 spaces)

**Comment by dmlap** [comment]
Obsoleted by #321
